grayskull/pypi/pypi.py

```diff
@@ -95,14 +102,15 @@ def _get_sdist_metadata(self, sdist_url: str, name: str) -> dict:
         :return: sdist metadata
         """
         temp_folder = mkdtemp(prefix=f"grayskull-{name}-")
-        pkg_name = sdist_url.split("/")[-1]
+        pkg_name = sdist_url.split("/")[-3]
+        print(f"This is the pkg_name as fed in the get_sdist_metadata method: {pkg_name}")
         path_pkg = os.path.join(temp_folder, pkg_name)
-
-        PyPi._download_sdist_pkg(sdist_url=sdist_url, dest=path_pkg)
+        print(f"This is the path_pkg: {path_pkg}")
+        PyPi._download_sdist_pkg(sdist_url=sdist_url, name= name, dest=path_pkg)
         if self._download:
             self.files_to_copy.append(path_pkg)
         log.debug(f"Unpacking {path_pkg} to {temp_folder}")
-        shutil.unpack_archive(path_pkg, temp_folder)
+        shutil.unpack_archive(path_pkg, temp_folder, format="gztar")
```

Comment:
We can't force this. Make sure `path_pkg` is properly encoding an extension shutil recognizes. If it doesn't we need to provide a mapping to figure it out and default to None so they can give it a try.

Reply:
Done!!